src/librbd/CMakeLists.txt

```diff
@@ -188,12 +188,12 @@ endif()
 if(WITH_EVENTTRACE)
   add_dependencies(rbd_internal eventtrace_tp)
 endif()
-target_link_libraries(rbd_internal PRIVATE
+target_link_libraries(rbd_internal
   ceph_immutable_object_cache_lib
   osdc)
 
 if(WITH_RBD_RWL)
-  target_link_libraries(rbd_internal PRIVATE
+  target_link_libraries(rbd_internal
```

Comment:
IIUC, `libpmemobj.h` is offered by `pmem::pmemobj`, why do you want to expose `pmem::pmem` as well?

Reply:
@tchaikov Because we add our test file` test_mock_ReplicatedWriteLog.cc`  in `unittest_librbd` in `test/librbd/CMakeLists.txt`. The file `test_mock_ReplicatedWriteLog.cc` need `libpmemobj.h`.  While `unittest_librbd` can't access `pmem::pmemobj` if set` rbd_internal` as PRIVATE here.  `unittest_librbd `depends on `rbd_internal `, `rbd_internal  `has `pmem::pmemobj `as private.

Reply:
Can't we just add `pmem::pmemobj` to `unittest_librbd`?

Reply:
@dillaman This is another option. Look like you prefer this option. Okay. I'll use this way.

Reply:
@dillaman Done

Reply:
@yison @dillaman linking `unittest_librbd` against `pmem::pmemobj` would be that last resort, i think. as unittest_librbd does not test `pmem::pmemobj` explicitly, this dependency is involved by rbd , or rbd_internal to be specific. so ideally, we should fix the dependency on `rbd_internal` or whatever library that pulls in `pmem::pmemobj` linkage or the header inclusion dependency.

i will try compile unittest_librbd locally with PMDK enabled to reproduce this issue and get back to you.

Reply:
@yison @dillaman could you help test and review #34362 ? i think it's a better fix.

Reply:
@tchaikov lgtm